src/utils.py

```diff
@@ -31,8 +31,8 @@ def git_remote_branches(url):
 
 def _git_remote_refs(url):
     cmd = "git ls-remote --heads %s" % url
-    output = local(cmd, capture=True)
-    return [line.split() for line in output.splitlines()]
+    output = local(cmd, capture=True)['stdout']
```

Comment:
`stdout` is always split in lines? Not all commands do work on multiple lines e.g. binary output

Reply:
There is an option in Fabric I haven't implemented (many actually), but this one is called [`linewise`](https://github.com/mathiasertl/fabric/blob/master/sites/docs/usage/env.rst#linewise)

> Forces buffering by line instead of by character/byte, typically when running in parallel mode. This option is implied by `env.parallel` -- even if linewise is False, if parallel is True then linewise behavior will occur.

which is in contrast to parallel-ssh that *only* [gives you lines](https://github.com/ParallelSSH/parallel-ssh/blob/5f4a4137e58cd7d7716e8193b4de12e52ffb7832/pssh/clients/native/single.py#L401-L406). That `output_buffer` comes from [this function](https://github.com/ParallelSSH/parallel-ssh/blob/379b8713fa79592b7e45ee4c9b8715efa36e525e/pssh/native/_ssh2.pyx#L33-L67) that is reading the bytes directly until it encounters a line separator then yielding. At that point it's quite close to the C++ wrapper.

I really can appreciate the need to occasionally have access to the raw bytes as they're coming out but we're just not doing that anywhere. In the cases where we were doing that we ended up reading by line anyway.

There may be gremlins here with pssh+parallel code, I'm not sure.

Reply:
By gremlins I mean interleaving of output across parallel processes. I haven't noticed any but that could be because it's well handled elsewhere

Reply:
>  IIRC Fabric tarnished the output anyway e.g. by inserting IP addresses of the server on every line

we have more control over this now. I disabled the PSSH logger so it wouldn't do the exact same thing. Once support is merged you can change the backend with an environment variable and decide what sort of output you would prefer